lib/cgroup/cgroup.go

```diff
@@ -171,7 +172,7 @@ func (s *Service) Place(sessionID string, pid int) error {
 // readPids returns a slice of PIDs from a file. Used to get list of all PIDs
 // within a cgroup.
 func readPids(path string) ([]string, error) {
-	f, err := os.Open(path)
+	f, err := utils.OpenFile(path, false)
```

Comment:
It's not clear at the call-site what the extra boolean is for. What do you think about a separate func for when we don't want to follow symlinks? Something like `OpenFileNoSymlinks(path)` (then we could just use the standard `os.Open` when we want the default behavior.

Reply:
I suppose we could as long as the symlink is verbose in naming `OpenFileAllowingSymlinks`.  The goal in having it be a parameter is that the developer must make a choice in what parameter is provided.

Let me know what you think would be less likely to have a copy paste error, I think either could be fine as long as the name is verbose.

Reply:
> The goal in having it be a parameter is that the developer must make a choice in what parameter is provided.

Well, the developer doesn't have to make a choice if they don't know that we've created our own set of utilities that replace what the standard library offers. Most Go devs are not going to reach for these by default (especially because it's buried in a `utils` package with other misc and unrelated code).

I think it'd be better to use the standard library function when you want the default behavior of the standard library, and to use our overrides when you need something different.

Reply:
The goal is that "need something different" should be our default.  In general we should not allow the following of symlinks.  Do you have any thoughts on how we can make using this pattern a more default behavior?

Reply:
+1 for "OpenFileNoSymlink" (or some other variation) instead of OpenFile(string, bool). Because OpenFile already exists is not the same as os.Open, I suppose we ought to keep both.

A way to get this through the codebase is to replace all usages, as at least when doing maintenance it should draw attention. Other than that, we could try and put together a linter of sorts.

Reply:
Updated the function naming.  I don't quite want to remove the ability to use `os.OpenFile` yet.  I am hopeful that this can still become the default when you need to open a read only file.

That said, we do need to keep in mind symlinks when writing too